libraries/helpers.rb

```diff
@@ -8,17 +8,29 @@ class CloudPassage
     # windows_configuration_variables and linux_configuration_variables,
     # respectively
     class ConfigHelper
-      def initialize(agent_key, opt = {})
-        @agent_key = agent_key
-        initialize_params(opt)
-        @win_conf = WinConfig.new(opt)
-        @lin_conf = LinConfig.new(opt)
+      def initialize(opt = {})
+        # Chef::Log.warn("Initialization Data: #{opt}")
+        final_params = initialize_params(opt)
+        @win_conf = WinConfig.new(final_params)
+        @lin_conf = LinConfig.new(final_params)
       end
 
       def initialize_params(params = {})
-        params.each do |key, val|
-          instance_variable_set("@#{key}", val) unless params.empty?
+        final_config = blend_configs(params[:base_config],
+                                     params[:databag_config],
+                                     params[:encrypted_databag_config])
+        final_config.each do |key, val|
+          instance_variable_set("@#{key}", val)
         end
+        final_config
+      end
+
+      def blend_configs(base, databag, edatabag)
+        final_config = {}
+        base.each { |key, val| final_config["#{key}"] = val }
```

Comment:
You likely don't need to do string interpolation of your keys - Ruby should just do The Right Thing.

```
base.each { |key, val| final_config[key] = val }
```

You may even be able to just call "#merge" on these hashes:

```

def blend_configs(base, databag, edatabag)
   base.merge(databag).merge(edatabag)
end
```

Stupid ruby tricks!